src/ray/protobuf/common.proto

```diff
@@ -312,14 +312,17 @@ message TaskSpec {
   string concurrency_group_name = 24;
   // Whether application-level errors (exceptions) should be retried.
   bool retry_exceptions = 25;
+  // A serialized exception list that serves as an allowlist of frontend-language
+  // exceptions/errors that should be retried.
+  bytes serialized_retry_exception_allowlist = 26;
```

Comment:
want to call out the backwards-incompatibility of this change. Perhaps not that big of a deal given we don't support rolling upgrades to the cluster.